lib/protocol/bep_extensions.go

```diff
@@ -145,7 +157,15 @@ func (f FileInfo) IsEmpty() bool {
 	return f.Version.Counters == nil
 }
 
-// IsEquivalent checks that the two file infos represent the same actual file content,
+func (f FileInfo) IsEquivalent(other FileInfo) bool {
+	return f.isEquivalent(other, false, false, 0)
+}
+
+func (f FileInfo) IsEquivalentOptional(other FileInfo, ignorePerms bool, ignoreBlocks bool, ignoreFlags uint32) bool {
```

Comment:
What is the reason for this function? Or differently: Is there ever a situation where a different flag than local change needs to be ignored or where the local change flag must not be ignored?

Reply:
I added the parameterless `IsEquivalent` because that's often the one you want, and not having to pass false, false, etc to get the default behavior is nice. The `...Optional` is the old one with all the parameters for when you need that.

Having the `ignoreFlags` as a parameter is necessary. The local change flag should not be ignored in equality comparisons unless you are doing the comparison in a receive only folder. The flow of flag bits is that the scanner gets a set of bits that it's supposed to set on newly discovered/changed files, and this set of bits also goes into the ignore set. This set of bits is zero during the whole process for send-receive and send-only folders and is `LocalFlagReceiveOnlyChanged` for receive-only folders.